### PR TITLE
WIP: Google pay android implementation

### LINKED ISSUE #9: Add Google Pay for Android
Add Google Pay support for Android similar to how Apple Pay is supported for iOS.

https://stripe.com/docs/google-pay

And add a guide similar to https://github.com/stripe/stripe-react-native/blob/master/docs/apple-pay.md

### PR DESCRIPTION
**Pull request by arekkubaczkowski**
resolves #9 

- [x] create android bindings
- [x] create js methods and types
- [x] adapting the code to stripe-android breaking changes
- [x] write documentation

### CODE REVIEW

**Comment by ustkirill** [comment]
Hi guys!

Your package very useful, but I need google pay also, I tried use other packages, but it's not work. Do you have any time estimate for this PR?

Can I help with anything? :)

**Comment by jaaywags** [comment]
> Can I help with anything? :)

Yes, I am happy to help too anywhere I can. Thanks!

File: docs/GooglePay.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import { useGooglePay } from 'stripe-react-native';
+
+function CheckoutScreen() {
+  const [initialized, setInitialized] = useState(false);
+
+  // continued from above
+
+  return (
+    <Screen>
+      <Button
```
Comment (thorsten-stripe) [inline docs/GooglePay.md:96]: Let's link to https://developers.google.com/pay/api/android/guides/brand-guidelines for reference on how to style the button

File: example/src/screens/GooglePayScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    if (error) {
+      Alert.alert(error.code, error.message);
+      return;
+    }
+    setInitialized(true);
+  };
+
+  return (
+    <PaymentScreen onInit={initialize}>
+      <Button
```
Comment (thorsten-stripe) [inline example/src/screens/GooglePayScreen.tsx:97]: Style button based on https://developers.google.com/pay/api/android/guides/brand-guidelines

Comment (thorsten-stripe) [reply example/src/screens/GooglePayScreen.tsx:97]: image

File: example/src/screens/GooglePayScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        variant="primary"
+        title="Pay with Google Pay"
+      />
+      <View style={styles.row}>
+        <Button
+          disabled={!initialized}
+          onPress={createPaymentMethod}
+          loading={loading}
+          variant="primary"
+          title="Crete payment method"
```
Comment (thorsten-stripe) [inline example/src/screens/GooglePayScreen.tsx:110]: Use basic Gpay button without text https://developers.google.com/pay/api/android/guides/brand-guidelines

image

**Review by thorsten-stripe — CHANGES_REQUESTED** [review]
Some nits regarding the button

**Review by thorsten-stripe — APPROVED** [review]
LGTM, thanks

**Comment by fredriksormo** [comment]
When will google pay be available in the expo version of this package?

**Comment by thorsten-stripe** [comment]
> When will google pay be available in the expo version of this package?

It will be available from Expo SDK 43 onward which is slotted for release in October.

**Pull request merged**